yylocky/CITS5206-Group3
# issue14,15,16-Features for view page

# Change Summary 📝

### Features implemented on `view_all_workallocation_page`:

- **Role-Based Access Control:**
  * `HOS` and `Admin` can view **all** works.
  * `HOD` can view **only** their department's works.
  * `Staff` can view **only** their individual work.

- **Department Display Logic:**
  * When related to leaves, the department displayed is the user's department.
  * Otherwise, the work's assigned department is displayed.

- **Table Enhancements:**
  * **Filter:** Added functionality to filter each column.
  * **Sorting:** Columns can be sorted alphabetically or numerically.
  * **Export:** Users can export the table data as a CSV report.

- **Possible Improvement:**
  * Consider introducing an "Other" department, given that `HOS` and `Admin` actually don't belong to any PMC department, currently for simplicity they are associated with the Physics department (`dept id=1`) .

---

## Change Form 📋

- [x] The pull request title references the issue number.
- [x] I have conducted a self-review of my own code.
- [x] I have commented on my code, particularly in complex areas.
- [x] Tests (if applicable) have been added or updated.

---

## Background Context 🖼️

- **Since "view_workload" pagename has been used when @kylelhk developing issue25 for filtered workallocation result, the original view_workload has been changed to view_all_workload, thus navbar, routes and related files have been adjusted accordingly.**

- **A more comprehensive testing dataset has been created and uploaded to share with members for testing. refer to the login credential as below:**
![Screen Shot 2023-09-19 at 04 05 02](https://github.com/yylocky/CITS5206-Group3/assets/99595619/d2e46579-0d8a-46a2-9d3b-54615ada157b)

- **Test been conducted:**
 1.  Login as hos,hod,admin,staff respectively, only authorised data is displayed.
 2.  If work category=LEAVE, the department shows the user's department, otherwise task's deparment
 3.  filter function works as expected
 4.  sort function works as expected
 5.  export csv file successfully


---

## Reviewers 🕵️

(Optional) Suggest reviewers that might be interested in this PR.
